This patch is based on portions of upstream commit 8b208053d plus the whole
of upstream commit fccf0180, it was adapted for use in the Debian package
by Peter Michael Green.
9
commit 8b208053fac17321121323f22110d4cb337f8365
Author: Jason Ozias <jason.g.ozias@gmail.com>
Date:   Thu Jul 21 20:46:04 2022 -0400

    dependency updates and resulting necessary changes

commit fccf01808a7e243e5ee7e868558ba9924b9e19ca
Author: Jason Ozias <jason.g.ozias@gmail.com>
Date:   Thu Jul 21 20:57:20 2022 -0400

    fixed sysinfo issue

Index: vergen/src/feature/si.rs
===================================================================
--- vergen.orig/src/feature/si.rs
+++ vergen/src/feature/si.rs
@@ -19,7 +19,7 @@ use {
 use {
     crate::{config::VergenKey, feature::add_entry},
     getset::{Getters, MutGetters},
-    sysinfo::{ProcessorExt, System, SystemExt},
+    sysinfo::{CpuExt, System, SystemExt},
 };
 
 /// Configuration for the `VERGEN_SYSINFO_*` instructions
@@ -199,7 +199,7 @@ pub(crate) fn configure_sysinfo(instruct
                 config.cfg_map_mut(),
                 VergenKey::SysinfoCpuVendor,
                 system
-                    .processors()
+                    .cpus()
                     .get(0)
                     .map(|processor| processor.vendor_id().to_string()),
             );
@@ -219,9 +219,9 @@ pub(crate) fn configure_sysinfo(instruct
                 VergenKey::SysinfoCpuName,
                 Some(
                     system
-                        .processors()
+                        .cpus()
                         .iter()
-                        .map(ProcessorExt::name)
+                        .map(CpuExt::name)
                         .collect::<Vec<&str>>()
                         .join(","),
                 ),
@@ -233,7 +233,7 @@ pub(crate) fn configure_sysinfo(instruct
                 config.cfg_map_mut(),
                 VergenKey::SysinfoCpuBrand,
                 system
-                    .processors()
+                    .cpus()
                     .get(0)
                     .map(|processor| processor.brand().to_string()),
             );
@@ -244,7 +244,7 @@ pub(crate) fn configure_sysinfo(instruct
                 config.cfg_map_mut(),
                 VergenKey::SysinfoCpuFrequency,
                 system
-                    .processors()
+                    .cpus()
                     .get(0)
                     .map(|processor| processor.frequency().to_string()),
             );
@@ -262,7 +262,9 @@ pub(crate) fn configure_sysinfo(_instruc
 
 #[cfg(all(feature = "si", not(target_os = "windows"), not(target_os = "macos")))]
 fn check_user(process: &Process, user: &User) -> bool {
-    *user.uid() == process.uid
+    use sysinfo::ProcessExt;
+
+    Some(user.id()) == process.user_id()
 }
 
 #[cfg(all(feature = "si", target_os = "windows"))]
Index: vergen/Cargo.toml
===================================================================
--- vergen.orig/Cargo.toml
+++ vergen/Cargo.toml
@@ -66,7 +66,7 @@ version = "0.4.0"
 optional = true
 
 [dependencies.sysinfo]
-version = "=0.23"
+version = "0.24.7"
 optional = true
 default-features = false
 
